File: 04-2-One2NineAppWithRoom_kt2/app/src/main/java/com/example/one2nineappwithroom_kt2/game/GameConfig.kt
package com.example.one2nineappwithroom_kt2.game


class GameConfig {
    private val configurations: MutableList<List<Number>> = mutableListOf()
    private var current: Int = 0

    init {
        configurations.add(getNumberNumber())
//        configurations.add(getNumberRoman())
        configurations.add(getNumberEnglish())
        reset()
    }

    public fun reset() {
        current = 0
    }

    private fun getNumberNumber(): List<Number> {
        return listOf(
            Number(1, "1"),
            Number(2, "2"),
            Number(3, "3"),
            Number(4, "4"),
            Number(5, "5"),
            Number(6, "6"),
            Number(7, "7"),
            Number(8, "8"),
            Number(9, "9")
        )
    }

    private fun getNumberRoman(): List<Number> {
        return listOf(
            Number(1, "I"),
            Number(2, "II"),
            Number(3, "III"),
            Number(4, "IV"),
            Number(5, "V"),
            Number(6, "VI"),
            Number(7, "VII"),
            Number(8, "VIII"),
            Number(9, "IX")
        )
    }

    private fun getNumberEnglish(): List<Number> {
        return listOf(
            Number(1, "one"),
            Number(2, "two"),
            Number(3, "three"),
            Number(4, "four"),
            Number(5, "five"),
            Number(6, "six"),
            Number(7, "seven"),
            Number(8, "eight"),
            Number(9, "nine")
        )
    }

    fun getNextConfiguration(): List<Number> {
        val ret = configurations[current].toMutableList()
        ret.shuffle()
        current++
        return ret
    }

    fun hasNext(): Boolean {
        return current < configurations.size
    }
}

File: flutter_projects/04-1-one2nineapp_fl/lib/game/gameConfig.dart
import 'dart:math';

import 'number.dart';

class GameConfig {
  List<List<Number>> configurations = [];
  int current = 0;

  GameConfig() {
    configurations.add(getNumberNumber());
    // configurations.add(getNumberRoman());
    // configurations.add(getNumberEnglish());
    reset();
  }

  void reset() {
    current = 0;
  }

  List<Number> getNumberNumber() {
    return [
      Number(value: 1, label: '1'),
      Number(value: 2, label: '2'),
      Number(value: 3, label: '3'),
      Number(value: 4, label: '4'),
      Number(value: 5, label: '5'),
      Number(value: 6, label: '6'),
      Number(value: 7, label: '7'),
      Number(value: 8, label: '8'),
      Number(value: 9, label: '9')
    ];
  }

  List<Number> getNumberRoman() {
    return [
      Number(value: 1, label: 'I'),
      Number(value: 2, label: 'II'),
      Number(value: 3, label: 'III'),
      Number(value: 4, label: 'IV'),
      Number(value: 5, label: 'V'),
      Number(value: 6, label: 'VI'),
      Number(value: 7, label: 'VII'),
      Number(value: 8, label: 'VIII'),
      Number(value: 9, label: 'IX')
    ];
  }

  List<Number> getNumberEnglish() {
    return [
      Number(value: 1, label: 'one'),
      Number(value: 2, label: 'two'),
      Number(value: 3, label: 'three'),
      Number(value: 4, label: 'four'),
      Number(value: 5, label: 'five'),
      Number(value: 6, label: 'six'),
      Number(value: 7, label: 'seven'),
      Number(value: 8, label: 'eight'),
      Number(value: 9, label: 'nine')
    ];
  }

  List<Number> getNextConfiguration() {
    var ret = List<Number>.from(configurations[current]);
    ret.shuffle(Random());
    current++;
    return ret;
  }

  bool hasNext() {
    return current < configurations.length;
  }
}
